Address TRAC-555

CanOfBees opened the pull request:
**JIRA Ticket**: [TRAC-555](https://jira.lib.utk.edu/browse/TRAC-555)

# What does this Pull Request do?
This PR updates the UTK_ir_etds_post_process stylesheet to programmatically add mods:recordInfo[@displayLabel='Submission'] as well as programmatically update a child element of same to reflect changes to the MODS datastream by an end user. Additionally, it addresses mods:originInfo and updates the children and their values appropriately.

We need to add submission information to different elements in the MODS datastream. This PR addresses those needs.

# What's new?
1) template rules for mods:originInfo
1a) add a mods:dateCreated on initial datastream creation
1b) on subsequent modifications to the datastream, update the value of mods:dateCreated
2) template rules for mods:recordInfo[@displayLabel='Submission']
2a) conditionally check for mods:recordInfo..., if not present then create the element and children with appropriate values
2b) on subsequent modifications to the datastream, update the value of mods:recordChangeDate to reflect the date/time of the changes.

# How should this be tested?
* acquire my repo and branch and place it in the sanctuary of your freshly destroyed and rebuilt TRACE vm
* `sudo cp the-path-to-my-repo/post_process_transforms/UTK_ir_etds_post_process.xsl /var/www/drupal/sites/all/modules/islandora_xml_forms/builder/self_transforms/`
* **verify** that the *Thesis MODS Form* has an **self-transform** association linking it to the UTK_ir_etds_post_process stylesheet.
* Create a new record in the GUI with the standard MODS template; i.e. Thesis MODS Form.
* Review the output to verify that the following XPaths are populated with w3cdtf-lookin' date values:
  * mods:originInfo/mods:dateCreated[@encoding='w3cdtf'] (= '2017-01-21T21:05:52-5:00'; e.g.)
  * mods:recordInfo[@displayLabel='Submission']/mods:recordCreationDate[@encoding='w3cdtf'] (= '2017-01-21T20:55:32-5:00'; e.g.)
  * mods:recordInfo[@displayLabel='Submission']/mods:recordChangeDate[@keyDate='yes'][@encoding='w3cdtf']
    * this element/XPath is repeatable!
* Note any problems with these XPaths and continue!
* 'Edit' the MODS datastream, using the same form, and repeat the XPath checks.
  * mods:dateCreated should reflect the time of the **most recent change!**
  * mods:recordCreationDate should reflect the time of the **initial submission *not* subsequent edits!**
  * mods:recordDateChange should repeat from where-ever to there-ever, reflecting the times the MODS datastream was manipulated via the form!

1. Option 1:
	* Import the Form
	* Associate it with a content model
	* Apply any additional / related transforms
	* Create a new record and select the newly associated form
	* Edit that record to see if the form still behaves correctly
2. Option 2:
	* vagrant destroy
	* vagrant up

# Interested parties
@markpbaggett @DonRichards @pc37utn @cdeaneGit @robert-patrick-waltz

markpbaggett commented:
@CanOfBees Looking at this now.

markpbaggett commented:
Looking at this now.  

- /originInfo/dateCreated[@encoding="w3cdtf"] seems to populate fine.
- /recordInfo stuff doesn't get in.  Looks like this is due to your template match.  This looks like it's based on the old, invalid DGI way of doing things, but I could be wrong?

CanOfBees commented:
@markpbaggett thanks for catching that error. I'm working on updating and testing a commit now.

CanOfBees commented:
@markpbaggett, when you have a moment please re-check this PR.

markpbaggett commented:
Hmmm.  Is this working for you?  For some reason, this still isn't working for me.  I tried clearing cache after overwriting that file.  Anything else I should try?

CanOfBees commented:
Let me retest.

CanOfBees commented:
Okay, mods:physicalDescription may have been a bad choice for the template match as it looks like it may be added after the self-transform has run. Updated the XPath template match to mods:originInfo[mods:dateCreated], which is added automatically from the form.

markpbaggett commented:
So, yeah, I'm convinced the last problem has to do with when this runs. 

That being said, interestingly the originInfo stuff doesn't work after last commit??

CanOfBees commented:
@markpbaggett  -

thanks for your patience. I updated the template rules for mods:originInfo and moved the existential check for mods:recordInfo into a more appropriate place. I've tested and it seems like it's working* (on my machine (whatever that means :))). 

\* working = originInfo's children are populating correctly and the existential check is working correctly to add recordInfo and its children. Editing the MODS dsid updates the appropriate values, too.

markpbaggett commented:
Looks good
